ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+            /// \param shapeDimIndex Dimension index
+            /// \param dimName Name to assign on this dimension
+            virtual void set_name_for_dimension(Place::Ptr place,
+                                                size_t shapeDimIndex,
+                                                const std::string& dimName);
+
+            ///// Topology Editing  /////
+
+            /// \brief Cut immediately before this place and assign this place as new input; prune
+            /// all nodes that don't contribute to any output.
+            /// \param place New place to be assigned as input
+            /// \param newNameOptional Optional new name assigned to this input place
+            virtual void cut_and_add_new_input(Place::Ptr place,
+                                               const std::string& newNameOptional = "");
+
+            /// \brief Cut immediately after this place and assign this place as new output; prune
+            /// all nodes that don't contribute to any output.
+            /// \param place New place to be assigned as output
+            /// \param newNameOptional Optional new name assigned to this output place
+            virtual void cut_and_add_new_output(Place::Ptr place,
+                                                const std::string& newNameOptional = "");
+
+            /// \brief Assign this place as new output or add necessary nodes to represent a new
+            /// output.
+            ///
+            /// \param place Anchor point to add an output
+            /// \return new output place, may be the same as a given place
+            virtual Place::Ptr add_output(Place::Ptr place);
+
+            /// \brief Removes any sinks directly attached to this place with all inbound data flow
+            /// if it is not required by any other output.
+            /// \param place Model place
+            virtual void remove_output(Place::Ptr place);
+
+            /// \brief Replaces all existing outputs with new ones removing all data flow that is
+            /// not required for new outputs.
+            ///
+            /// \param outputs Vector with places that will become new outputs; may intersect
+            /// existing outputs.
+            /// \param outputs Array of new output places
+            virtual void override_all_outputs(const std::vector<Place::Ptr>& outputs);
+
+            /// \brief Modifies the graph to use new inputs instead of existing ones. New inputs
+            /// should completely satisfy all existing outputs.
+            /// \param inputs Array of new input places
+            virtual void override_all_inputs(const std::vector<Place::Ptr>& inputs);
+
+            /// \brief Leaves only subgraph that are defined by new inputs and new outputs.
+            /// \param inputs Array of new input places
+            /// \param outputs Array of new output places
+            virtual void extract_subgraph(const std::vector<Place::Ptr>& inputs,
```

Comment:
Should we return Ptr to the new InputModel?
From my point of view, extract sound like a sub-graph copy rather than inplace cutting.

Reply:
From my point of view, it is quite complicated requirement for frontends to support creation of 'subgraph' copy, so it should definitely do inplace cutting. Also this name looks consistent with 'onnx_editor::SubgraphExtractor::extract_subgraph' which also does inplace cut. @slyalin should it be renamed?

Reply:
@GlebKazantaev, all methods modify model in-place. Name is not ideal, agree. Do you have a better alternative?